internal/logs/workload_writer.go

```diff
@@ -0,0 +1,60 @@
+package logs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/project-flotta/flotta-device-worker/internal/workload"
+)
+
+type WorkloadWriter struct {
+	workloadName string
+	cancel       context.CancelFunc
+	target       *FifoLog
+	targetName   string
+	manager      *workload.WorkloadManager
+}
+
+func NewWorkloadWriter(workload string, manager *workload.WorkloadManager) *WorkloadWriter {
+	res := &WorkloadWriter{
+		workloadName: workload,
+		manager:      manager,
+	}
+	return res
+}
+
+func (ww *WorkloadWriter) SetTarget(name string, target *FifoLog) {
```

Comment:
It does not seem to be called in other context than right after WorkloadWriter is created; can't that be a part of the constructor?

Reply:
Want to make it simpler, we heavily use the constructor with a ton of params, and in this way, for me, it's a bit easier to understand. 

Also, this way I was also thinking on update transports in case of changes in the future.